src/charm.py

```diff
@@ -28,13 +35,20 @@
 CONFIG_FILE_NAME = "ausfcfg.conf"
 CONFIG_TEMPLATE_DIR = "src/templates/"
 CONFIG_TEMPLATE_NAME = "ausfcfg.conf.j2"
+CERTS_DIR_PATH = "/free5gc/support/TLS"  # Certificate paths are hardcoded in AUSF code
+PRIVATE_KEY_NAME = "ausf.key"
+CSR_NAME = "ausf.csr"
+CERTIFICATE_NAME = "ausf.pem"
+CERTIFICATE_COMMON_NAME = "ausf.sdcore"
 
 
 class AUSFOperatorCharm(CharmBase):
     """Main class to describe juju event handling for the SD-Core AUSF operator."""
 
     def __init__(self, *args) -> None:
         super().__init__(*args)
+        if not self.unit.is_leader():
```

Comment:
If you deploy 1 unit of 1 application it will for sure be the leader, what else can it be?

Reply:
That's a good question, and I am trying to figure out the answer.
There might be no issue, but my doubt is the following:
Is this unit created as a leader or is there a time when there is no leader (in the case of multiple units this can happen) and if there is a time with no leader, could we have a race condition here so our charm crashes because there is not yet a leader unit?

Reply:
The whole point of this here is to prevent people from scaling the app. If somebody creates multiple units, all of them (except the leader) will go to error state

Reply:
I have been doing some testing, and it seems you are right, we can assume that the unit is going to be leader from the beginning, for example I found this line in the logs `unit-udr-0: 05:55:35 INFO juju.worker.leadership udr/0 promoted to leadership of udr` which is happening before any charm specific things.